tools/serverpod_cli/bin/create/create.dart

```diff
@@ -25,6 +25,16 @@ Future<void> performCreate(
   String template,
   bool force,
 ) async {
+  // Check whether the project name is valid or not
+  for (int i = 0; i < name.length; i++) {
```

Comment:
This code block actually works. But this is a little expensive way to check if the name is valid. I can give you a hint so that you can approach it. Instead of using for loop use regex and check if the name matches that regex or not. `'[a-z_][a-z0-9_]*'` This regex will help you in that. This reduces time complexity. So it will be much faster.

Reply:
Yes, a regex is a better solution. Project names should probably not start with an underscore though. So it can be just `'[a-z][a-z0-9_]*'`